satpy/tests/reader_tests/test_ici_l1b_nc.py

```diff
@@ -0,0 +1,432 @@
+
+    def test_drop_coords(self):
+        """Test drop coordinates."""
+        coords = "dummy"
+        data = xr.DataArray(
+            np.ones(10),
+            dims=('y'),
+            coords={coords: 0},
+        )
+        assert coords in data.coords
+        data = self.reader._drop_coords(data, coords)
+        assert coords not in data.coords
+
+    def test_get_dataset_return_none_if_data_not_exist(self):
+        """Tes get dataset return none if data does not exist."""
+        dataset_id = {'name': 'unknown'}
+        dataset_info = {'file_key': 'non/existing/data'}
+        dataset = self.reader.get_dataset(dataset_id, dataset_info)
+        assert dataset is None
+
+    @patch('satpy.readers.ici_l1b_nc.IciL1bNCFileHandler._calibrate_bt')
+    def test_get_dataset_does_not_calibrate_if_not_desired(
+        self,
+        mocked_calibrate,
+    ):
+        """Test get dataset does not calibrate if not desired."""
+        dataset_id = {'name': 'lon_pixels_horn_1'}
+        dataset_info = {
+            'name': 'lon_pixels_horn_1',
+            'file_type': 'nc_ici_l1b_rad',
+            'file_key': 'cached_longitude',
+            'standard_name': 'longitude',
+            'n_horns': 0,
+            'modifiers': (),
+        }
+        dataset = self.reader.get_dataset(dataset_id, dataset_info)
+        assert dataset.dims == ('y', 'x')
+        mocked_calibrate.assert_not_called()
+        assert isinstance(dataset, xr.DataArray)
+
+    @patch('satpy.readers.ici_l1b_nc.IciL1bNCFileHandler._perform_orthorectification')  # noqa: E501
+    def test_get_dataset_orthorectifies_if_orthorect_data_defined(
+        self,
+        mocked_orthorectification,
+    ):
+        """Test get dataset orthorectifies if orthorect data is defined."""
+        dataset_id = {'name': 'lon_pixels_horn_1'}
+        dataset_info = {
+            'name': 'lon_pixels_horn_1',
+            'file_type': 'nc_ici_l1b_rad',
+            'file_key': 'cached_longitude',
+            'orthorect_data': 'data/navigation_data/delta_longitude',
+            'standard_name': 'longitude',
+            'n_horns': 0,
+            'modifiers': (),
+        }
+        self.reader.get_dataset(dataset_id, dataset_info)
+        mocked_orthorectification.assert_called_once_with(
+            ANY,  # the data
+            dataset_info['orthorect_data'],
+        )
+
+    @patch('satpy.readers.ici_l1b_nc.IciL1bNCFileHandler._perform_calibration')
+    def test_get_dataset_handles_calibration(
+        self,
+        mocked_calibration,
+    ):
+        """Test get dataset handles calibration."""
+        dataset_id = {'name': 'ici_1'}
+        dataset_info = {
+            'name': 'ici_1',
+            'file_type': 'nc_ici_l1b_rad',
+            'file_key': 'data/measurement_data/ici_radiance_183',
+            'coordinates': ['lat_pixels_horn_1', 'lon_pixels_horn_1'],
+            'n_183': 0,
+            'n_horns': 0,
+            'chan_index': 0,
```

Comment:
Can we test that these functions (calibration, orthorectification) perform as expected instead of just testing that they are called?

Reply:
Changed so these tests are not mocked and the return value is now asserted.